hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/ApacheHttpClientHealthMonitor.java

```diff
@@ -0,0 +1,33 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hadoop.fs.azurebfs.services;
+
+
+public final class ApacheHttpClientHealthMonitor {
```

Comment:
I don't think we need a separate class for this

Reply:
Wanted to encapsulate static methods related to apache usability in one class, hence not keeping in AbfsRestOperation, but having a separate static class. Open to removing it, but just wanted to discuss if we should keep or remove it.

Reply:
What I feel is we still should keep it different from AbfsRestOperation. but I feel there are a lot of new classes related to Apache Cleint. We can try to move them to a common file and have multiple classes defined. If we can convert some into internal classes that would also be good.

Just a thought.

Reply:
Have removed the Monitor class, and have kept logic in the AbfsApacheHttpClient. I am not inclined to keep the new classes as internal for following reasons:
1. the classes are big and internalizing them will lead to big code files
2. static-inner class would be difficult to handle with mockito.
3. Each of the new class are encapsulating some logic and are not just object description.

Would be insightful to know your thoughts on this.